joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -1417,13 +1444,34 @@ void JointTrajectoryController::preempt_active_goal()
   }
 }
 
-void JointTrajectoryController::set_hold_position()
+std::shared_ptr<trajectory_msgs::msg::JointTrajectory>
+JointTrajectoryController::set_hold_position()
 {
-  trajectory_msgs::msg::JointTrajectory empty_msg;
-  empty_msg.header.stamp = rclcpp::Time(0);
+  // Command to stay at current position
+  trajectory_msgs::msg::JointTrajectory current_pose_msg;
+  current_pose_msg.header.stamp =
+    rclcpp::Time(0.0, 0.0, get_node()->get_clock()->get_clock_type());  // start immediately
+  current_pose_msg.joint_names = params_.joints;
+  current_pose_msg.points.push_back(state_current_);
+  current_pose_msg.points[0].velocities.clear();
+  current_pose_msg.points[0].accelerations.clear();
+  current_pose_msg.points[0].effort.clear();
+  if (has_velocity_command_interface_)
+  {
+    current_pose_msg.points[0].velocities.resize(
+      dof_, 0.0);  // ensure no velocity (PID will fix this)
```

Comment:
read a little better
```suggestion
    // ensure no velocity (PID will fix this)
    current_pose_msg.points[0].velocities.resize(dof_, 0.0);
```